### ISSUE TITLE
Properties set null after reloading

### ISSUE THREAD

**Issue by Nosfistis** [typeorm/typeorm#5424]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
After partially updating an entity through save, its fields are not reloaded, if they are not part of the update.
E.g. in the following entity, if I update using `{id: 1, isActive: true}`, then the entity is persisted but reloaded as `{id: 1, isActive: true, description: null}`, even though the description was and still is populated in the database.
```Typescript
@Entity()
export class Scenario extends BaseEntity {

  @ManyToOne(() => Project, project => project.scenarios, { nullable: false, onUpdate: 'CASCADE', onDelete: 'CASCADE' })
  project?: Project;

  @RelationId((scenario: Scenario) => scenario.project)
  @Column()
  projectId: number;

  @OneToOne(() => State, { eager: true, cascade: ['remove'], onUpdate: 'CASCADE' })
  @Type(() => State)
  @JoinColumn()
  state: State;

  @Column({ length: 255, nullable: true })
  description?: string;

  @Column({ default: true })
  isActive?: boolean;

}
```

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by pjungwir** [comment]
This is still relevant. Typeorm does not reload the entity on `save` even if you pass `{reload: true}` (when doing a SQL `UPDATE`). See https://github.com/typeorm/typeorm/issues/8836 for more details.

Ideally Typeorm would use `RETURNING` for Postgres and only do a second query when your RDBMS doesn't support that (e.g. MySQL).

**Comment by DillonSadofsky** [comment]
Agreed, still relevant.  And like I point out above, typeorm already fully retrieves the row from the database.  The round trip is already spent, the cost is paid, so just use that data to update the entity if reload is true.
